plugin/providers/phpipam/data_source_phpipam_address.go

```diff
@@ -53,7 +54,19 @@ func dataSourcePHPIPAMAddressRead(d *schema.ResourceData, meta interface{}) erro
 			return err
 		}
 	default:
-		return errors.New("No valid combination of parameters found - need one of address_id, ip_address, or subnet_id and (description|hostname|custom_field_filter)")
+    id := d.Id()
+    if len(id) > 0 {
+      address_id, err := strconv.Atoi(id)
+      if err != nil {
+        return err
+      }
+		  out[0], err = c.GetAddressByID(address_id)
+		  if err != nil {
+			  return err
+		  }
+    } else {
```

Comment:
I don't get the point of this change. Can you please add a comment why this check is needed?